### DISCUSSION TITLE
lcm sdxl support

### DISCUSSION THREAD

**Discussion by frankyifei** [lllyasviel/Fooocus#899] [Ideas]
It is supported in comfyui now, lcm lora works with any sdxl model. very fast inference speed. but cfg needs to be set around 1.

**Comment by aolko** [comment]
Here's the workflow
https://gist.github.com/comfyanonymous/b948fd68aacc65c2ed37d743d0964f27

Here are the loras
https://huggingface.co/collections/latent-consistency/latent-consistency-models-loras-654cdd24e111e16f0865fba6

**Comment by joanorsky** [comment]
Is it working on fooocus?

**Reply by Pdonor** [reply]
It seems to work now! Download the lora and set the sampler to "LCM"

**Reply by KINGLIFER** [reply]
Sampler? Not lora?

**Reply by kienerj** [reply]
For anyone else ending up here:

1. [Download](https://civitai.com/models/195519?modelVersionId=219782) the lora for sdxl 
2. Move it to the models/loras folder of fooocus
3. Activate Advanced options
4. In the model tab select the LCM lora and set weight to 1 (not sure what other weights have as an effect)
5. In the Advanced Tab, Activate the Developer Debug mode
6. Scroll down and you will se an Option for "Sampler". Set it to "lcm"
7. Scroll a bit further down to "Forced Overwrite of Sampling Step", there you can set sampling steps to 3 or 4

**Comment by joanorsky** [comment]
The lora weight is 1 .. right? Having some mixed results...

**Reply by joanorsky** [reply]
ok.. it's working. It even faster with the SSD1B model...

**Comment by frankyifei** [comment]
Is it possible to let users to set cfg large than 1.0 to take advantage of negative prompt?

**Reply by joanorsky** [reply]
you can set the guidance scale from 1 to 2 (not more or it will go wild like a mad kangaroo)...

**Reply by frankyifei** [reply]
I prefer 1.2-1.5 but currently in Fooocus, if you choose Extreme Speed, it disables the option to set CFG. Also I think 12 steps give me better results, the default for Extreme Speed is 8, which is OK in many cases but lacking some details. Anyway, all these can be overcome by using customized config.txt to set. I think there should be a easier way in the gui to let user make the decision.

**Reply by joanorsky** [reply]
Advanced -> Developer Debug Mode -> Forced Overwrite of Sampling Step -> Set your steps here

**Comment by mirek190** [comment]
> It is supported in comfyui now, lcm lora works with any sdxl model. very fast inference speed. but cfg needs to be set around 1.

Is already build in and called extreme speed
